### ISSUE TITLE
DataONE library not processing UTF-8 correctly

### ISSUE THREAD

**Issue by elopatin-uc3** [CDLUC3/mrt-doc#51]
UTF-8 encoding in Dataone data causes the Merritt processing library to raise an exception
This is an example object that triggers the error

== UTF-8 (Algoritmos de Classificação.docx)
Job ID: jid-b0bf7d0b-a45e-4145-9a51-02e7b501fd0b
Primary ID: ark:/13030/m5130pnh
Local ID: doi:10.15146/R3TC7F
Version: 2
Filename: mrt-delete.txt
Object title: Testing J48 and KNN Algorithms with Hypothyroid dataset
Object creator: Zotelli dos Santos, Wanderley JosÃ©
Object date: 2017
Submission date: 2018-04-30T14:03:16-07:00
Completion date: 2018-04-30T14:03:30-07:00
Metacat Registration Status: failure
Status: COMPLETED

 - [ ] determine if this is initiated with Dash created manifest or if this is Merritt

**Comment by elopatin-uc3** [comment]
Stack trace:

Exception:org.cdlib.mrt.dataone.content.VersionContent-retrieveComponent:247 INVALID_OR_MISSING_PARM[Program req
uired parm is missing or invalid] VersionContent: component file not found: - componentFile=/apps/dpr2/ingest_ho
me/queue/bid-a281f047-4884-43c9-bd65-242c2821054f/jid-b0bf7d0b-a45e-4145-9a51-02e7b501fd0b/producer/Wanderley - 
Algoritmos de Classificação.docx
org.cdlib.mrt.dataone.content.VersionContent-retrieveComponent:247 INVALID_OR_MISSING_PARM[Program required parm
 is missing or invalid] VersionContent: component file not found: - componentFile=/apps/dpr2/ingest_home/queue/b
id-a281f047-4884-43c9-bd65-242c2821054f/jid-b0bf7d0b-a45e-4145-9a51-02e7b501fd0b/producer/Wanderley - Algoritmos
 de Classificação.docx
        at org.cdlib.mrt.dataone.create.DataOneResource.getFileComponent(DataOneResource.java:245)
        at org.cdlib.mrt.dataone.create.DataOneResource.fillEntry(DataOneResource.java:218)
        at org.cdlib.mrt.dataone.create.DataOneResource.addLine(DataOneResource.java:171)
        at org.cdlib.mrt.dataone.create.DataOneResource.buildResourceTable(DataOneResource.java:135)
        at org.cdlib.mrt.dataone.create.DataOneResource.<init>(DataOneResource.java:96)
        at org.cdlib.mrt.dataone.create.DataOneHandler.initialize(DataOneHandler.java:309)
        at org.cdlib.mrt.dataone.create.DataOneHandler.<init>(DataOneHandler.java:231)
        at org.cdlib.mrt.dataone.create.DataOneHandler.getDataOneHandler(DataOneHandler.java:104)
        at org.cdlib.mrt.ingest.handlers.HandlerDataONE.handle(HandlerDataONE.java:245)
        at org.cdlib.mrt.ingest.handlers.HandlerDataONE.handle(HandlerDataONE.java:111)
        at org.cdlib.mrt.ingest.IngestManager.submit(IngestManager.java:617)
 (Mark Reyes - May 8, 2018)

**Comment by elopatin-uc3** [comment]
This may not be an encoding issue in Merritt.  The payload data seems to be mangled coming into Merritt.  Dash?  Storage?

Wanderley - Algoritmos de Classifica??o.docx (Mark Reyes - May 9, 2018)

**Comment by elopatin-uc3** [comment]
investigate Dash vs Merritt
 (Marisa Strong - May 21, 2018)

**Comment by elopatin-uc3** [comment]
Still investigating.  However, it looks as though the GMN can not handle UTF-8...

2018-05-23 22:49:57 ERROR    root exception_handler 16334 140576869771008     term.py(1301)
2018-05-23 22:49:57 ERROR    root exception_handler 16334 140576869771008     Type: <type 'exceptions.UnicodeEncodeError'>
2018-05-23 22:49:57 ERROR    root exception_handler 16334 140576869771008     Value: 'ascii' codec can't encode character u'\u0327' in position 65: ordinal not in range(128)


I'll send a note to Roger Dahl (Mark Reyes - May 23, 2018)

**Comment by elopatin-uc3** [comment]
I believe that the Merritt/Dash problem has been addressed.  The failure we saw with object 

https://merritt.cdlib.org/m/ark%253A%252F13030%252Fm5130pnh/2

is due to the files being mangled, probably due to a migration issue.
This issue will be closed.  However, I'll leave the ticket open until we resolve the issue with UTF-8 and the GMN (note has been sent to M. Servilla and R. Dahl)
 (Mark Reyes - May 23, 2018)
